sdk/digitaltwins/digital-twins-core/test/public/testDigitalTwins.spec.ts

```diff
@@ -0,0 +1,939 @@
+        updatedTwin.body.AverageTemperature,
+        42,
+        "Unexpected TemperatureUnit result from updateDigitalTwin()."
+      );
+    } catch (error) {
+      errorWasThrown = true;
+      assert.include(error.message, `There is no digital twin instance that exists with the ID`);
+    } finally {
+      await deleteDigitalTwin(digitalTwinId);
+      await deleteModels();
+    }
+    should.equal(errorWasThrown, false, "Error was thrown");
+  });
+
+  it("update digital twin invalid conditions", async function() {
+    const digitalTwinId = recorder.getUniqueName(
+      "digitalTwin",
+      "update-digitaltwin-invalid-conditions"
+    );
+
+    await setUpModels();
+    await deleteDigitalTwin(digitalTwinId);
+
+    const buildingTwin = {
+      $metadata: {
+        $model: BUILDING_MODEL_ID
+      },
+      AverageTemperature: 68,
+      TemperatureUnit: "Celsius"
+    };
+    const patch = [
+      {
+        op: "replace",
+        path: "/AverageTemperature",
+        value: 42
+      }
+    ];
+    await client.upsertDigitalTwin(digitalTwinId, JSON.stringify(buildingTwin));
+
+    let options: DigitalTwinsDeleteOptionalParams = {
+      ifMatch: "XXX"
+    };
+    let errorWasThrown = false;
+    try {
+      await client.updateDigitalTwin(digitalTwinId, patch, options);
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(
+        error.message,
+        `Invalid If-Match header value. Allowed value(s): If-Match: {etag} or If-Match: *`
+      );
+    } finally {
+      await deleteDigitalTwin(digitalTwinId);
+      await deleteModels();
+    }
+    should.equal(errorWasThrown, true, "Error was not thrown");
+  });
+
+  it("update digital twin not existing", async function() {
+    const digitalTwinId = recorder.getUniqueName("digitalTwin", "update-digitaltwin-not-existing");
+
+    await setUpModels();
+    await deleteDigitalTwin(digitalTwinId);
+
+    const patch = [
+      {
+        op: "replace",
+        path: "/AverageTemperature",
+        value: 42
+      }
+    ];
+
+    let errorWasThrown = false;
+    try {
+      await client.updateDigitalTwin(digitalTwinId, patch);
+    } catch (error) {
+      errorWasThrown = true;
+      assert.include(error.message, `There is no digital twin instance that exists with the ID`);
+    } finally {
+      await deleteDigitalTwin(digitalTwinId);
+      await deleteModels();
+    }
+    should.equal(errorWasThrown, true, "Error was not thrown");
+  });
+
+  it("query digital twin", async function() {
+    const digitalTwinId = recorder.getUniqueName("digitalTwin", "query-digitaltwin");
+
+    await setUpModels();
+    await deleteDigitalTwins();
+
+    const buildingTwin = {
+      $metadata: {
+        $model: BUILDING_MODEL_ID
+      },
+      AverageTemperature: 68,
+      TemperatureUnit: "Celsius"
+    };
+    await client.upsertDigitalTwin(digitalTwinId, JSON.stringify(buildingTwin));
+
+    // Wait for the service to be ready
+    await delay(5000);
```

Comment:
Hey @zolvarga!

As this code might closely reflect the user's code, "upserting and querying". Do you want to check with someone from the digital-twins service if this delay is expected?

Reply:
This is a know delay by the service team. The time needed to populate the Twin object on the server. 
Also, the scenario is not really a user scenario.